server.go

```diff
@@ -490,17 +1296,29 @@ func (s *server) handleQuery(querymsg interface{}, state *peerState) {
 		}
 
 		// TODO(oga) if too many, nuke a non-perm peer.
-		if s.handleAddPeerMsg(state,
-			newOutboundPeer(s, msg.addr, msg.permanent, 0)) {
+		p := s.addPeer(msg.addr)
+		go func() {
+			conn, err := btcdDial("tcp", msg.addr)
+			if err != nil {
+				// Only log the error if we're not forcibly shutting down.
+				if atomic.LoadInt32(&s.shutdown) == 0 {
+					srvrLog.Errorf("can't accept connection: %v",
```

Comment:
Not accepting, establishing.

Also, I see the same code in 3 different places.  This seems ripe for a function.  Maybe something like:

```Go
// establishConn attempts to establish a connection with the passed peer.  It
// will typically need to be run as a goroutine to avoid blocking the caller.
func (s *server) establishConn(p *peer.Peer) {
	conn, err := btcdDial("tcp", p.Addr())
	// ....
	p.Connect(conn)
	return nil
}
```

Then, the code here would be:

```Go
p := s.addPeer(msg.addr)
go s.establishConn(p)
```